Libraries/Opc.Ua.Server/Subscription/Subscription.cs

```diff
@@ -2268,19 +2283,33 @@ private void ConditionRefresh(List<IEventMonitoredItem> monitoredItems, uint mon
         /// <summary>
         /// Sets the subscription to durable mode.
         /// </summary>
-        public ServiceResult SetSubscriptionDurable(uint lifeTimeInHours, out uint revisedLifeTimeInHours)
+        public ServiceResult SetSubscriptionDurable(uint maxLifetimeCount)
         {
             lock (m_lock)
             {
-                // set default
-                revisedLifeTimeInHours = 0;
+                if (!m_supportsDurable)
+                {
+                    Utils.LogError("SetSubscriptionDurable requested for subscription with id {0}, but no IMonitoredItemQueueFactory that supports durable queues was registered", m_id);
+                    return StatusCodes.BadInternalError;
```

Comment:
is this really the right status code here?

Reply:
BadNotSupported instead?